### ISSUE TITLE
Makefile does not account for cross-compiling

### ISSUE THREAD

**Issue by rkeene** [ray2501/tcl-lmdb#11]
https://github.com/ray2501/tcl-lmdb/blob/992c681f8039165a8bdacd05ca8da9f19aa7ba8d/Makefile.in#L152-L161

This assumes that because one is building ON Linux that one is building FOR Linux, this does not hold true when cross-compiling.

**Comment by ray2501** [comment]
Yes, it is try to fix glibc pthread robust mutex support (glibc < 2.12) case on and for Linux build for myself. So I do not account for cross-compiling. Any idea for this?

**Comment by rkeene** [comment]
The LMDB source itself seems to deal with it:

https://github.com/ray2501/tcl-lmdb/blob/992c681f8039165a8bdacd05ca8da9f19aa7ba8d/generic/mdb.c#L309-L328

**Comment by ray2501** [comment]
I add this because it results in build failed in the Ubuntu 10.04 environment (glibc 2.11, pthread function is missing or other things).

However, I do not know how to check this in this moment because I updated my environment to newer Ubuntu.

**Comment by rkeene** [comment]
Maybe changing the glibc check to be 2.11 instead of 2.4 is sufficient ?

**Comment by ray2501** [comment]
LMDB check glibc < 2.10 when I write code but current LMDB code base check is correct.

I think remove this workaround or instead of check glibc 2.4 is OK. Do you suggest remove the workaround or instead of check glibc 2.4?

**Comment by rkeene** [comment]
Compiling it for Android still fails with the robust check failing, so maybe something better needs to be done as part of autoconf.  I'll submit a patch.
